edb/server/protocol/auth_ext/data.py

```diff
@@ -80,3 +80,126 @@ def __repr__(self) -> str:
+    acr_values_supported: Optional[list[str]]
+    subject_types_supported: list[str]
+    id_token_signing_alg_values_supported: list[str]
+    id_token_encryption_alg_values_supported: Optional[list[str]]
+    id_token_encryption_enc_values_supported: Optional[list[str]]
+    userinfo_signing_alg_values_supported: Optional[list[str]]
+    userinfo_encryption_alg_values_supported: Optional[list[str]]
+    userinfo_encryption_enc_values_supported: Optional[list[str]]
+    request_object_signing_alg_values_supported: Optional[list[str]]
+    request_object_encryption_alg_values_supported: Optional[list[str]]
+    request_object_encryption_enc_values_supported: Optional[list[str]]
+    token_endpoint_auth_methods_supported: Optional[list[str]]
+    token_endpoint_auth_signing_alg_values_supported: Optional[list[str]]
+    display_values_supported: Optional[list[str]]
+    claim_types_supported: Optional[list[str]]
+    claims_supported: Optional[list[str]]
+    service_documentation: Optional[str]
+    claims_locales_supported: Optional[list[str]]
+    ui_locales_supported: Optional[list[str]]
+    claims_parameter_supported: Optional[bool]
+    request_parameter_supported: Optional[bool]
+    request_uri_parameter_supported: Optional[bool]
+    require_request_uri_registration: Optional[bool]
+    op_policy_uri: Optional[str]
+    op_tos_uri: Optional[str]
+
+    def __init__(self, **kwargs):
+        for field in dataclasses.fields(self):
+            setattr(self, field.name, kwargs.get(field.name))
+
+    def __str__(self) -> str:
+        return self.issuer
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"issuer={self.issuer!r}, "
+            f"authorization_endpoint={self.authorization_endpoint!r}, "
+            f"token_endpoint={self.token_endpoint!r}, "
+            f"jwks_uri={self.jwks_uri!r}, "
+            f"response_types_supported={self.response_types_supported!r}, "
+            f"subject_types_supported={self.subject_types_supported!r}, "
+            "id_token_signing_alg_values_supported="
+            f"{self.id_token_signing_alg_values_supported!r})"
+        )
+
+
+@dataclasses.dataclass
+class OAuthAccessTokenResponse:
+    """
+    Access Token Response.
+    https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.4
+    """
+
+    access_token: str
+    token_type: str
+    expires_in: int
+    refresh_token: str
+
+    def __init__(self, **kwargs):
+        for field in dataclasses.fields(self):
+            setattr(self, field.name, kwargs.get(field.name))
+
+    def __str__(self) -> str:
+        return self.access_token
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"access_token={self.access_token!r}, "
+            f"token_type={self.token_type!r}, "
+            f"expires_in={self.expires_in!r}, "
+            f"refresh_token={self.refresh_token!r})"
+        )
```

Comment:
Any best practices about `__str__` and `__repr__` for things that contain potentially sensitive data like this?

Reply:
Let's mask the secrets here if `EDGEDB_DEBUG_EXT_AUTH` (`edb.common.debug.flags.ext_auth`) is set - so that we can safely log the response object when needed.

Reply:
I wonder if it's even worth having a nice `__repr__` for these classes since the only non-sensitive data is pretty useless for disambiguation (`token_type`, `expires_in`). WDYT?

Reply:
True! So we can just use dataclass's repr for ext::auth debugging then. Makes sense!

Reply:
I'm suggesting turning off the dataclass `__repr__` via `@dataclasses.dataclass(repr=False)`. That wouldn't give us access to a nicer debug output though, even during debugging, right?

Reply:
@fantix 👀 [`d4f6cde` (#6013)](https://github.com/edgedb/edgedb/pull/6013/commits/d4f6cde44e3db95bd2744f8e29aad258f67de953)

this is what I was thinking